### PR TITLE
PYTHON-4868 Generate server tests using shrub.py

### PR DESCRIPTION
**Pull request by blink1073**
- Break up the test matrix to cover all server versions, python versions, platforms, topologies, and auth/ssl with as little repetition as possible
- Break up sync and async tests on MacOS and Windows

Before: 1026 tasks across 39 build variants
After: 495 tasks across 37 build variants

image

### CODE REVIEW

**Comment by blink1073** [comment]
I'll need to update the GitHub PR aliases once this is merged.

**Comment by blink1073** [comment]
I need to add coverage...

File: .evergreen/config.yml [hunk 1]
```diff
@@ -2399,6 +2400,470 @@ axes:
...
+    SSL: ssl
+    TEST_SUITES: default_async
+    PYTHON_BINARY: C:/python/32/Python313/python.exe
+    SKIP_CSOT_TESTS: "true"
+- name: test-win32-py3.9-noauth-nossl-sync
+  tasks:
+    - name: .sharded_cluster
+  display_name: Test Win32 py3.9 NoAuth NoSSL Sync
+  run_on:
+    - windows-64-vsMulti-small
```
Comment (NoahStapp) [inline .evergreen/config.yml:2847]: We run 32-bit python on a 64-bit machine?

**Review by NoahStapp — CHANGES_REQUESTED** [review]
One clarification, otherwise looks great! This will be so nice for testing going forward

File: .evergreen/config.yml (hunk 1 above)
Comment (blink1073) [reply .evergreen/config.yml:2847]: 32-bit versions of Python.  No one runs 32 bit machines anymore lol.

Comment (blink1073) [reply .evergreen/config.yml:2847]: But some apps still require 32 bit

Comment (NoahStapp) [reply .evergreen/config.yml:2847]: Just wanted to make sure, makes sense lol

**Comment by blink1073** [comment]
evergreen retry

**Comment by blink1073** [comment]
evergreen retry

**Pull request merged**